### ISSUE TITLE
"login:forward, user" hook is not triggered when logging in with hybridauth

### ISSUE THREAD

**Issue by oseg** [arckinteractive/elgg_hybridauth#28]
Side effect: plugins like "login_redirector" cannot work

Suggested workaround:

In views/default/resources/hybridauth/authenticate.php at line 136, replace the block
```
	// Profile for this provider exists
	if (!elgg_is_logged_in()) {
		$user_to_login->elgg_hybridauth_login = 1;
		login($user_to_login);
		system_message(elgg_echo('hybridauth:login:provider', array($provider->getName())));
		forward($forward_url);
	}
```
by
```

	// Profile for this provider exists
	if (!elgg_is_logged_in()) {
		$user_to_login->elgg_hybridauth_login = 1;
                try {
                        login($user_to_login);
                } catch(LoginException $e){
                        register_error($e->getMessage());
                        forward(REFERER);
                }
                             
                $user = $user_to_login;
                $forward_source = null;
                
                // elgg_echo() caches the language and does not provide a way to change the language.
                // @todo we need to use the config object to store this so that the current language
                // can be changed. Refs #4171
                if ($user->language) {
                        $message = elgg_echo('hybridauth:login:provider', array($provider->getName()), $user->language);
                } else {
                        $message = elgg_echo('hybridauth:login:provider', array($provider->getName()));
                }

                // clear after login in case login fails
                if ($session) $session->remove('last_forward_from');

                $params = array('user' => $user, 'source' => $forward_source);
                $forward_url = elgg_trigger_plugin_hook('login:forward', 'user', $params, $forward_url);
                
		system_message($message);
		forward($forward_url);
	}
```
This workaround is triggering the hook from a view, but usually, the "login:forward, user" hook is triggered by the 'login' action. I don't know if it is safe to do things this way...

**Comment by hypeJunction** [comment]
Feel free to make actual pull requests. Neither @beck24 nor I will have time for these any time soon.

**Comment by oseg** [comment]
fixed by a8cd2fd00ca78c4b32fa048082c773a3a816ccda
